feat(STN-125):  hero 1 styles

silverli opened the pull request:
# READY FOR REVIEW

## Summary

[JIRA card](https://sparkbox.atlassian.net/browse/STN-125)

[Figma Designs](https://www.figma.com/file/toNZnFLadtpNpG5dPzNGWz/%E2%9A%99%EF%B8%8F-Colorful-Theme---Working?node-id=3446%3A0)

Creates static styles for hero option 1. Also tweaks, just slightly, the hero pattern UI html.

### Screenshots
**Mobile**
<img width="578" alt="Screen Shot 2020-02-20 at 8 35 59 PM" src="https://user-images.githubusercontent.com/12848168/74997280-bb3ac180-5423-11ea-88df-1d5220a75000.png">
<img width="574" alt="Screen Shot 2020-02-20 at 8 36 06 PM" src="https://user-images.githubusercontent.com/12848168/74997281-bbd35800-5423-11ea-9b1c-79853788adf4.png">

**Desktop**
![Screen Shot 2020-02-20 at 8 15 21 PM](https://user-images.githubusercontent.com/12848168/74997274-ba099480-5423-11ea-996c-ed427ee30666.png)
<img width="946" alt="Screen Shot 2020-02-20 at 8 35 32 PM" src="https://user-images.githubusercontent.com/12848168/74997276-baa22b00-5423-11ea-8d1f-5047609bf6fb.png">


## Steps to Test
1. Create a basic page.
2. Add a paragraph type "hero," or a couple with different sized images and different lengths of text. Check the page and styles at different breakpoints and see that everything looks ok.
3. In the humsci_basic folder, run `npm run test` and make sure everything passes.

## PR Checklist
- [PR Checklist](https://gist.github.com/sherakama/0ba17601381e3adbe0cad566ad4d80a5)
- [Sparkbox PR Checklist](../docs/SparkboxPRChecklist.md)

Commit:
feat(STN-125): hero option 1 pattern styles

- implements css grid hero option 1 styles with pseudo element color blocks
- adds border button mixin and styling for buttons and links
- alters some of the pattern UI html twig markup for the option 1 hero

kaseybon commented on docroot/themes/humsci/humsci_basic/src/scss/components/_pattern.hero.scss:
```diff
@@ -0,0 +1,136 @@
+// looks very bad if constrained in 2 or 1 column layout
+// also, not sure how to get this to be outside .hb-page-width
```
The question of how to place this is going to be tackled in [STN-195](https://sparkbox.atlassian.net/browse/STN-195)

silverli replied:
ok great! thank you

kaseybon commented on docroot/themes/humsci/humsci_basic/src/scss/components/_pattern.hero.scss:
```diff
@@ -0,0 +1,136 @@
+
+    a {
+      display: block;
+      // TODO: make this into a border button mixin?
```
Let's go ahead and do that

kaseybon commented on docroot/themes/humsci/humsci_basic/src/scss/components/_pattern.hero.scss:
```diff
@@ -0,0 +1,136 @@
+        &:hover,
+        &:focus {
+          box-shadow: none;
+          // what's the hover effect here?
```
We'll need to figure this out before merging. The text color darkening is not accessible.

silverli replied:
Merani added designs that I implemented

kaseybon commented on docroot/themes/humsci/humsci_basic/src/scss/tools/_mixins.general.scss:
```diff
@@ -31,6 +39,17 @@
 
+// Adds a background box pseudo element
+// You can choose the color and height
+// Remember to add position: relative to the parent element
```
Excellent bit of documentation for something that would be easily forgotten!

kaseybon reviewed (changes requested):
This is looking good but I left a few comments in the code. There is also an issue with the mint colored block that is a pseudo element of the text area. I think we'll want this to be a pseudo element on the `hb-hero-overlay` element so we have more control over it's positioning in relation to the rest of the hero. When the a visitor is not logged in the color block lays on top of the masthead. This is happening at both large and small screen sizes.

<img width="928" alt="Screen Shot 2020-02-24 at 7 06 36 AM" src="https://user-images.githubusercontent.com/2486846/75151418-a2a50280-56d4-11ea-95d7-6a9fd2796f37.png">

silverli commented on docroot/themes/humsci/humsci_basic/src/scss/settings/_variables.general.scss:
```diff
@@ -24,6 +24,7 @@ $hb-screen-margins: (
 $hb-gutter-width: 48px;
 $hb-layout-max-width: 1200px;
 $hb-sidebar-width: 22.5%;
+$hb-hero-width: 300px;
```
I think I can delete this now, not using anymore

kaseybon replied:
I didn't see anything in the code base that's using it so we can go ahead and delete.

silverli commented on docroot/themes/humsci/humsci_basic/src/scss/tools/_mixins.general.scss:
```diff
@@ -16,6 +24,32 @@
+// Adds a background box pseudo element
+// You can choose the color and height
+// Remember to add position: relative to the parent element
+@mixin psuedo-background-box($color: $su-color-driftwood, $height: 100%, $width: 100%) {
```
might delete this too since it's not used anymore?

silverli replied:
actually jk it is getting used once

silverli commented on docroot/themes/humsci/humsci_basic/templates/page.html.twig:
```diff
@@ -20,6 +20,30 @@
   </section>
 </header>
 
+<div class="hb-hero-overlay hb-hero-overlay--1">
```
TODO delete

kaseybon commented on docroot/themes/humsci/humsci_basic/patterns/hero-text-overlay/hero-text-overlay.html.twig:
```diff
@@ -1,13 +1,22 @@
-{% set attributes = attributes.addClass('hero-text-overlay') %}
+{% set attributes = attributes.addClass('hb-hero-overlay') %}
+{% set attributes = attributes.addClass('hb-hero-overlay--1') %}
```
If we have multiple classes we need to use an array to set the classes. You can use the horizontal card as an example.

kaseybon commented on docroot/themes/humsci/humsci_basic/src/scss/tools/_mixins.text.scss:
```diff
@@ -147,3 +147,28 @@
   }
 }
+
+@mixin hb-border-button {
```
When you rebase there is a new `_mixins.buttons.scss` partial. This should live there instead.

kaseybon commented on docroot/themes/humsci/humsci_basic/templates/page.html.twig:
```diff
@@ -20,6 +20,30 @@
+<div class="hb-hero-overlay hb-hero-overlay--1">
+  {# These empty divs are used for styling the color blocks layered on the hero #}
+  <div class="hb-hero-overlay__grey-block"></div>
+  <div class="hb-hero-overlay__secondary-color-block"></div>
```
Can we some how scope these or make them dependent on the `hb-hero-overlay--1` modifier?

kaseybon committed:
fix(STN-125): run postcss on compiled CSS theme files

Commit:
feat(STN-125): adds our supported browsers to .browserslist config

- if you run `npx browserslist` you get out the supported browsers that we will autoprefix for. This is more than we stated we support, but it doesnt hurt to support the top 1% of modern browsers either:

and_chr 78
and_ff 68
and_qq 1.2
and_uc 12.12
android 76
baidu 7.12
chrome 78
chrome 77
chrome 76
chrome 49
edge 18
edge 17
firefox 70
firefox 69
firefox 68
ie 11
ie_mob 11
ios_saf 13.2
ios_saf 13.0-13.1
ios_saf 12.2-12.4
kaios 2.5
op_mini all
op_mob 46
opera 64
opera 63
safari 13
safari 12.1
safari 5.1
samsung 10.1
samsung 9.2
